From c9a3fbc4832c8510bac3e0c65c2412af76554b9b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juan=20Manuel=20L=C3=B3pez=20Carrillo?=
 <juanmanuellopezcarrillo@gmail.com>
Date: Tue, 14 Jul 2026 01:59:05 +0200
Subject: [PATCH 065/143] arm64: dts: sun55i-a523: add the CPU clock controller

clock-controller@8817000 node (allwinner,sun55i-a523-cpu-ccu) with its input
clocks (hosc/losc/iosc + pll-periph0-600m of the main CCU). Not wired into
cpu@N yet (read-only clock validation first).
---
 arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi b/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
index 031bdabcf..1b195f564 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
@@ -1160,5 +1160,14 @@ npu: npu@7122000 {
 			resets = <&mcu_ccu RST_BUS_MCU_NPU>;
 			power-domains = <&ppu PD_NPU>;
 		};
+
+		cpu_ccu: clock-controller@8817000 {
+			compatible = "allwinner,sun55i-a523-cpu-ccu";
+			reg = <0x08817000 0x4000>;
+			clocks = <&osc24M>, <&rtc CLK_OSC32K>, <&rtc CLK_IOSC>,
+				 <&ccu CLK_PLL_PERIPH0_600M>;
+			clock-names = "hosc", "losc", "iosc", "pll-periph0-600m";
+			#clock-cells = <1>;
+		};
 	};
 };
-- 
2.53.0

